src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -455,6 +459,41 @@ impl CatalogState {
         }
     }
 
+    pub(crate) fn pack_network_policy_update(
+        &self,
+        policy_id: &NetworkPolicyId,
+        diff: Diff,
+    ) -> Result<Vec<BuiltinTableUpdate<&'static BuiltinTable>>, Error> {
+        let policy = self.get_network_policy(policy_id);
+
+        let rules = Jsonb::from_str(&serde_json::to_string(&policy.rules).expect(""))
+            .map_err(|e| {
+                Error::new(ErrorKind::Unstructured(format!(
+                    "could not pack audit log update: {}",
+                    e
+                )))
+            })?
+            .into_row();
+        let rules = rules
+            .iter()
+            .next()
+            .expect("details created above with a single jsonb column");
```

Comment:
Not super sure about this being jsonb. It seemed to make sense, and there's some precedence for it, alternatively it might be cleaner to do an `mz_network_policy_rules` table.